mesonbuild/dependencies/hdf5.py

```diff
@@ -140,13 +143,6 @@ def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.
             elif Path(arg).is_file():
                 self.link_args.append(arg)
 
-        # If the language is not C we need to add C as a subdependency
-        if language != 'c':
-            nkwargs = kwargs.copy()
-            nkwargs['language'] = 'c'
-            # I'm being too clever for mypy and pylint
-            self.is_found = self._add_sub_dependency(hdf5_factory(environment, for_machine, nkwargs))  # pylint: disable=no-value-for-parameter
```

Comment:
I think this should be its own patch, so 3 in total.

Reply:
This can't be done in this PR, can it? Do I need to open a new one?

Reply:
You don't need to push to a new branch and open a new PR. :) You can just prepare the branch locally, and then use `git push -f` instead of `git push` to replace this PR with the new contents.

Reply:
the things you learn about git... Done.